tests/io/test.directory.js

```diff
@@ -61,85 +55,84 @@ describe('Directory.getFiles()', () => {
       return filePath === 'dir2/dir3/file3.txt';
     });
 
-    return myDirectory.getFiles()
-      .then((files) => {
-        const fileNames = Object.keys(files);
-        expect(fileNames).not.toContain('dir1/file1.txt');
-        expect(fileNames).not.toContain('dir2/file2.txt');
-        expect(fileNames).toContain('dir2/dir3/file3.txt');
-      });
+    const files = await myDirectory.getFiles();
+    const fileNames = Object.keys(files);
+    expect(fileNames).not.toContain('dir1/file1.txt');
+    expect(fileNames).not.toContain('dir2/file2.txt');
+    expect(fileNames).toContain('dir2/dir3/file3.txt');
   });
 });
 
 describe('Directory._getPath()', () => {
-  it('should reject if not a file that exists', () => {
+  it('should reject if not a file that exists', async () => {
     const myDirectory = new Directory('tests/fixtures/io/');
-    return myDirectory.getFiles()
-      .then(() => {
-        return myDirectory.getPath('whatever')
-          .then(unexpectedSuccess)
-          .catch((err) => {
-            expect(err.message).toContain(
-              '"whatever" does not exist in this dir.'
-            );
-          });
-      });
+
+    await myDirectory.getFiles();
+    try {
+      await myDirectory.getPath('whatever');
+      unexpectedSuccess();
```

Comment:
it looks that jest with a version >= 20 supports a `expect(...).rejects` assertion that we could use in out tests instead of the `try { ... } catch (...) { ... }`:

- https://facebook.github.io/jest/docs/en/expect.html#rejects

e.g. this one would become something like:

```js
    await expect(
      myDirectory.getPath('whatever')
    ).rejects.toThrow('"whatever" does not exist in this dir.');
```

and as a plus jest produce a very nice failure message when the rejected promise doesn't match the expected error message (or if it doesn't fail as expected).

There are a number of `try { ... } catch (...) { ... }` in this file that we can make shorter and cleaner applying this approach, and it could be worth given that we are already changing this test file.